### PR TITLE
Revert "Wildcard V2: <Link /> Manual migration (#30190)" to fix log-out

### PR DESCRIPTION
**Pull request by rafax**
This reverts commit PR #30190 (commit 04093189944bdb6ab1a4b2ae30d11ba020f7e46f) to fix logout link - culprit found by manual `git bisect`.

Broken on main: https://www.loom.com/share/fef5246cd9ee43099e21344d22cace22
Fixed in this PR: https://www.loom.com/share/a267a0ccb7804db4a2aba597cb84b914

### CODE REVIEW

**Review by oleggromov — APPROVED** [review]

File: client/web/src/nav/GlobalNavbar.test.tsx [hunk 1]
```diff
@@ -54,6 +55,8 @@ const PROPS: React.ComponentProps<typeof GlobalNavbar> = {
 }
 
 describe('GlobalNavbar', () => {
+    setLinkComponent(({ children, ...props }) => <a {...props}>{children}</a>)
+    afterAll(() => setLinkComponent(() => null)) // reset global env for other tests
```
Comment (valerybugakov) [inline client/web/src/nav/GlobalNavbar.test.tsx:59]: What's the purpose of adding this couple of lines?

Comment (rafax) [reply client/web/src/nav/GlobalNavbar.test.tsx:59]: These lines were removed in 04093189944 (that broke log-out), so I guess they were readded here (this PR is a clean `git revert 04093189944`).

Comment (rafax) [reply client/web/src/nav/GlobalNavbar.test.tsx:59]: This is the link to removal [diff](https://github.com/sourcegraph/sourcegraph/commit/04093189944bdb6ab1a4b2ae30d11ba020f7e46f#diff-a1bb43aeeab92cab231028f62c6abf372836e9f61528d8bb47bcb4b715bdbcba) - it takes a few seconds to scroll to the right place.

Comment (valerybugakov) [reply client/web/src/nav/GlobalNavbar.test.tsx:59]: @rafax, I see, thanks! Will address it in the migration PR then.

**Pull request merged**